BirthdayGreetingsTest/api.js

```diff
@@ -12,50 +12,45 @@ const app = express();
 const port = 3000;
 app.use(express.json());
 
-app.get("/birthdays/:date", async (req, res) => {
-  // Get all friend objects that have a birthday on the specified date
-  const date = req.params.date;
-  retrieveBirthdaysFromDB(date)
-    .then((friends) => {
-      res.send(friends);
-    })
-    .catch((err) => {
-      res.status(500).send(err);
-    });
+app.get("/birthdays", async (req, res) => {
+  const { startDate, endDate } = req.query;
+  if (startDate && endDate) {
+    // Get all friend objects that have a birthday within the specified range
+    retrieveBirthdaysWithinRangeFromDB(startDate, endDate)
+      .then((friends) => {
+        res.status(200).send(friends);
+      })
+      .catch((err) => {
+        res.status(err.status).send(err);
+      });
+  } else {
```

Comment:
What is a query parameter is not used? How should the system behave?

Reply:
@RichardWatkins1 It'll pass `undefined` into `retrieveBirthdaysFromDB()` then reject with a 400 status, should this be changed?